services/job_chat/job_chat.py

```diff
@@ -1,96 +1,145 @@
+        if hasattr(message, "usage"):
+            if message.usage.cache_creation_input_tokens:
+                logger.info(f"Cache write: {message.usage.cache_creation_input_tokens} tokens")
+            if message.usage.cache_read_input_tokens:
+                logger.info(f"Cache read: {message.usage.cache_read_input_tokens} tokens")
+
+        response_parts = []
+        for content_block in message.content:
+            if content_block.type == "text":
+                response_parts.append(content_block.text)
+            else:
+                logger.warning(f"Unhandled content type: {content_block.type}")
+
+        response = "\n\n".join(response_parts)
+
+        updated_history = history + [
+            {"role": "user", "content": content},
+            {"role": "assistant", "content": response},
+        ]
+
+        return ChatResponse(
+            content=response,
+            history=updated_history,
+            usage=message.usage.model_dump() if hasattr(message, "usage") else {},
+        )
 
-        if message.usage.cache_read_input_tokens:
-            print("CACHE :: READ :: {} Tokens".format(message.usage.cache_read_input_tokens))
 
-        response = []
+def main(data_dict: dict) -> dict:
+    """
+    Main entry point with improved error handling and input validation.
+    """
+    try:
+        data = Payload.from_dict(data_dict)
 
-        if response is None:
-            logger.error("An error occurred during during chat generation")
-        else:
-            # we need to unpack the contents into a flat string
-            # TODO: consider supporting richer response types
-            for r in message.content:
-                if r.type == "text":
-                    response.append(r.text)
+        config = ChatConfig(api_key=data.api_key) if data.api_key else None
+        client = ClaudeClient(config)
 
-            response = "\n\n".join(response)
+        result = client.generate(content=data.content, history=data_dict.get("history", []), context=data.context)
 
-            logger.info("response from model:")
-            logger.info("")
-            logger.info("\n" + response)
-            logger.info("")
-            logger.info("done")
+        return {"response": result.content, "history": result.history, "usage": result.usage}
 
-        history.append({"role": "user", "content": content})
-        history.append({"role": "assistant", "content": response})
+    except ValueError as e:
+        raise ApolloError(400, str(e), type="BAD_REQUEST")
 
-        # TODO maybe return usage ?
-        return {"response": response, "history": history}
+    except APIConnectionError as e:
+        raise ApolloError(
+            503, "Unable to reach AI service", type="CONNECTION_ERROR", details={"cause": str(e.__cause__)}
+        )
+    except AuthenticationError as e:
+        raise ApolloError(401, "Authentication failed. Please check your credentials.", type="AUTH_ERROR")
```

Comment:
I think we should leave this at `Authentication failed` for now. Because in Lightning, in the unlikely event that this occurs, there's nothing the user can do. There is no credential to check.

This is the benefit for sending an error type - later, if we want, we can add a more useful contextual message on the Lightning side, and a different one from the CLI side, or whatever.

But for now, let's keep the error messages as generic as we can